addons/website_sale/models/product.py

```diff
@@ -170,6 +177,25 @@ def _compute_parents_and_self(self):
             else:
                 category.parents_and_self = category
 
+    def get_product_count(self, categs_nbr_prod):
```

Comment:
As you suggested, you can try to move everything (that has no other use on the controller) to this method.

Also make it private.